odoo/osv/expression.py

```diff
@@ -868,11 +868,11 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                 (when available), or as an expanded [(left,in,child_ids)] """
             if not ids:
                 return [FALSE_LEAF]
-            left_model = left_model.with_context(active_test=False)
+            left_model = left_model.sudo().with_context(active_test=False)
```

Comment:
Suggestion: rename this variable as `left_model_sudo`, to make it crystal clear that it's sudoed everywhere.

Reply:
Good idea, agreed. Changed.